tests/device_tests/test_session_id_and_passphrase.py

```diff
@@ -0,0 +1,241 @@
+    assert isinstance(response, messages.Features)
+    session_id = response.session_id
+    assert len(session_id) == 32
+
+    # GetPublicKey requires passphrase and since it is not cached,
+    # Trezor will prompt for it.
+    xpub = _get_xpub(client, passphrase="A")
+    assert (
+        xpub
+        == "xpub6CekxGcnqnJ6osfY4Rrq7W5ogFtR54KUvz4H16XzaQuukMFZCGebEpVznfq4yFcKEmYyShwj2UKjL7CazuNSuhdkofF4mHabHkLxCMVvsqG"
+    )
+
+    # Call Initialize again, this time with the received session id and then call
+    # GetPublicKey. The passphrase should be cached now so Trezor must
+    # not ask for it again, whilst returning the same xpub.
+    response = client.call_raw(messages.Initialize(session_id=session_id))
+    assert isinstance(response, messages.Features)
+    xpub = _get_xpub(client, passphrase=None)
+    assert (
+        xpub
+        == "xpub6CekxGcnqnJ6osfY4Rrq7W5ogFtR54KUvz4H16XzaQuukMFZCGebEpVznfq4yFcKEmYyShwj2UKjL7CazuNSuhdkofF4mHabHkLxCMVvsqG"
+    )
+
+    # If we set session id in Initialize to None, the cache will be cleared
+    # and Trezor will ask for the passphrase again.
+    response = client.call_raw(messages.Initialize(session_id=None))
+    assert isinstance(response, messages.Features)
+    xpub = _get_xpub(client, passphrase="A")
+    assert (
+        xpub
+        == "xpub6CekxGcnqnJ6osfY4Rrq7W5ogFtR54KUvz4H16XzaQuukMFZCGebEpVznfq4yFcKEmYyShwj2UKjL7CazuNSuhdkofF4mHabHkLxCMVvsqG"
+    )
+
+    # Unknown session id is the same as setting it to None.
+    response = client.call_raw(messages.Initialize(session_id=b"X" * 32))
+    assert isinstance(response, messages.Features)
+    xpub = _get_xpub(client, passphrase="A")
+    assert (
+        xpub
+        == "xpub6CekxGcnqnJ6osfY4Rrq7W5ogFtR54KUvz4H16XzaQuukMFZCGebEpVznfq4yFcKEmYyShwj2UKjL7CazuNSuhdkofF4mHabHkLxCMVvsqG"
+    )
+
+
+@pytest.mark.skip_ui
+@pytest.mark.setup_client()
+def test_session_enable_passphrase(client):
+    # Let's start the communication by calling Initialize.
+    response = client.call_raw(messages.Initialize())
+    assert isinstance(response, messages.Features)
+    session_id = response.session_id
+    assert len(session_id) == 32
+
+    # Trezor will not prompt for passphrase because it is turned off.
+    xpub = _get_xpub(client, passphrase=None)
+    assert (
+        xpub
+        == "xpub6BiVtCpG9fQPxnPmHXG8PhtzQdWC2Su4qWu6XW9tpWFYhxydCLJGrWBJZ5H6qTAHdPQ7pQhtpjiYZVZARo14qHiay2fvrX996oEP42u8wZy"
+    )
+
+    # Turn on passphrase.
+    _enable_passphrase(client)
+
+    # The session id is unchanged, therefore we do not prompt for the passphrase.
+    response = client.call_raw(messages.Initialize(session_id=session_id))
+    xpub = _get_xpub(client, passphrase=None)
+    assert isinstance(response, messages.Features)
+    assert session_id == response.session_id
+    assert (
+        xpub
+        == "xpub6BiVtCpG9fQPxnPmHXG8PhtzQdWC2Su4qWu6XW9tpWFYhxydCLJGrWBJZ5H6qTAHdPQ7pQhtpjiYZVZARo14qHiay2fvrX996oEP42u8wZy"
```

Comment:
please use constants declared at start for the xpubs.
for instance, here it is not clear at first look that this is the same xpub as above